### DISCUSSION TITLE
Setting the SHIR value when deploying for a data flow

### DISCUSSION THREAD

**Discussion by turagittech** [Azure-Player/azure.datafactory.tools#378] [Q&A] [answered]
In the Data flow activity in a pipeline it requires an IR. With Secure data using Vnets and self hosted Integration runtimes it needs setting as part of delivery to production. There doesn't appear to be an option to make it a parameter/variable so it requires altering in the deployment. 
The IR value is several layers into the JSON for the pipeline. 
pipeline,PipelineName,activities."Activity Name".typeProperties.ifTrueActivities.integrationRuntime.referenceName,"runtime-name" 
I've tried a lot of variations on this string, but it's either not possible or I have slightly missed the mark. I do currently have a space in the activity name, not sure if that's a problem 

Does anyone have experience with setting this?

**Comment by turagittech** [comment]
I will add that jsonpath.com suggest this as correct jsonpath to reach the parameter, this also doesn't work 

PipelineName,activities[3].typeProperties."ifTrueActivities[0].typeProperties.integrationRuntime.referenceName

I forgot to add the error to my original post  ##[error]ADFT0010: Wrong path defined in config for object(path):

**Reply by NowinskiK** [reply]
Can I ask in the first instance, why you are trying to change "referenceName" in deep activity within a pipeline?
Could you also attach json file of this pipeline?

**Reply by turagittech** [reply]
I have SHIR running and they have different names. between dev and prod I cannot see how to use variable/parameters for the integration runtime  defined at this point. I modified the approach thinking it might be a path length issue. I've also added a simpler version. If there's a better way to reassign SHIR values when moving code from dev to production I'm all ears
[SQLRedicaseDataCopy.json](https://github.com/Azure-Player/azure.datafactory.tools/files/13444419/SQLRedicaseDataCopy.json)
 Oddly in testing I was doing last night PowerShell can access it when looking at the object outside of this apps code, but fails inside this app
[RunCDCRedicaseDataFlow.json](https://github.com/Azure-Player/azure.datafactory.tools/files/13444398/RunCDCRedicaseDataFlow.json)

**Reply by NowinskiK** [reply]
Consider having the same name for SHIR between environments (for the same reason you have identical names for pipelines, datasets, etc). Modifying pipeline activities is bad idea and you should avoid it as much as possible.

**Reply by turagittech** [reply, marked as answer]
I found a  Managed VNEt Azure Integration Runtime would work. I can name them the same so this is no longer required. An oversight on my part that Managed Vnet Azure IR could access Vnet Secured databases.

**Comment by turagittech** [comment]
Tried PowerShell Objects and followed the path which came back with this as an option for the path value
activities[3].typeproperties.IfTrueActivities.typeproperties.integrationruntime.referenceName
$a = gc .\FileName.json -Raw | ConvertFrom-Json
 foreach ($line in $a) {
>> $line.properties.activities[3].typeproperties.IfTrueActivities.typeproperties.integrationruntime.referenceName}

this also is not working
